x-pack/plugins/security_solution/common/detection_engine/schemas/request/perform_bulk_action_schema.ts

```diff
@@ -6,13 +6,28 @@
  */
 
 import * as t from 'io-ts';
-import { bulkAction, queryOrUndefined } from '../common/schemas';
+import { BulkAction, queryOrUndefined, bulkActionUpdate } from '../common/schemas';
 
-export const performBulkActionSchema = t.exact(
-  t.type({
-    query: queryOrUndefined,
-    action: bulkAction,
-  })
-);
+export const performBulkActionSchema = t.union([
+  t.exact(
+    t.type({
+      query: queryOrUndefined,
+      action: t.union([
+        t.literal(BulkAction.delete),
+        t.literal(BulkAction.disable),
+        t.literal(BulkAction.duplicate),
+        t.literal(BulkAction.enable),
+        t.literal(BulkAction.export),
+      ]),
+    })
+  ),
+  t.exact(
+    t.type({
+      query: queryOrUndefined,
+      action: t.literal(BulkAction.update),
+      updates: t.array(bulkActionUpdate),
```

Comment:
Perhaps for simplicity it'd make sense to call the property for the action's payload the same way as the action's name itself?

```ts
action: t.literal(BulkAction.update),
[BulkAction.update]: t.array(bulkActionUpdate),
```

Reply:
changed as well